src/E1_waterflow/E1_waterflow.cpp

```diff
@@ -105,23 +99,23 @@ void loop() {
   // process command
   if (RFSerial.available() > 0) {
     int i = 0;
+
     while (RFSerial.available()) {
       command[i] = RFSerial.read();
       Serial.print(command[i]);
       i++;
     }
 
-    debug(String(command), DEBUG);
-    int action = decode_received_packet(String(command), &valve, valves, numValves);
-    if (action != -1) {
-      take_action(&valve, action);
-      packet = make_packet(valve.id, false);
+    debug(String(command));
+    int8_t id = parseCommand(String(command));
```

Comment:
Same comment as on E1_coldflow.cpp - think it would be clearer if parseCommand only parsed the command, and the logic to take the action was somewhere else. Could allow certain IDs to be used to get an actuator, but then maybe have others just set a flag or change a parameter, without necessarily needing a whole actuator.

Reply:
the "actuator" is an abstract class. It can be anything we want it to be doesn't need to be a solenoid or heater. Actions like setting flags and changing parameters can be done w/in the context of an "actuator". Essentially they are virtual actuators. This prevents you from having divergences in the code. EVerything is handled w/ one case.